core/src/test/scala/unit/kafka/log/remote/RemoteIndexCacheTest.scala

```diff
@@ -540,6 +545,8 @@ class RemoteIndexCacheTest {
       "Failed to mark cache entry for cleanup after resizing cache.")
     TestUtils.waitUntilTrue(() => cacheEntry.isCleanStarted,
       "Failed to cleanup cache entry after resizing cache.")
+    TestUtils.waitUntilTrue(() => cacheEntry.isCleanFinished,
+      "Failed to finish cleanup cache entry after resizing cache.")
 
     // verify no index files on remote cache dir
     TestUtils.waitUntilTrue(() => !getIndexFileFromRemoteCacheDir(LogFileUtils.INDEX_FILE_SUFFIX).isPresent,
```

Comment:
@iit2009060 @hudeqi , adding an `isCleanFinished` flag just for test is not a good solution. In this case, could we catch the exception in the `getIndexFileFromRemoteCacheDir`?

Reply:
@hudeqi  Can you resolve the above comment ?

Reply:
@showuon  I have removed the `isCleanFinished` flag and catch the `Exception` in the method.
Please review it .